e2e/setup/packager.js

```diff
@@ -0,0 +1,70 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+const { spawn } = require("child_process");
+const { env } = require("../services");
+
+const packageFiles = async () => {
+  return new Promise((resolve, reject) => {
+    const ls = spawn("npm", ["run", "react:start"], {
```

Comment:
I thought about this for a while, do we want react:start or react:build here? Are we planning to test something during the middle of development? 

One issue is that we have to kill this process, instead if we relied on build, we could simply not care about the react-dev server during the teardown. WDYT?

Reply:
That's definitely a good point to discuss, I've also thought about it as I was writing this PR.

The reason I kept it as `"react:start"` is because I planned on dettaching the packager from the overall routine so that we can separate the packager server and the runner (like Kibana does).

**However, thinking about this one more time I think you're right and it's definitely worth going for `react:build` straightaway for the `test:e2e` command.**

I have just updated this PR so that we have a `test:e2e:server` and `test:e2e:runner` command for using `react:start` and a separate `test:e2e` which simply uses `build`.

Please lmk what you think.

(Great comment btw! 🙌 )